File: Permacore/backend/routes/shipment_routes.py
# Flask route in shipment_routes.py

from flask import Blueprint, request, jsonify
from ..services.shipment_services import ShipmentService

shipment_bp = Blueprint('shipment_bp', __name__)

@shipment_bp.route('/api/shipments', methods=['GET'])
def get_shipments():
    """Get all shipment records."""
    shipments = ShipmentService.get_all_shipments()
    return jsonify([shipment.to_dict() for shipment in shipments]), 200

@shipment_bp.route('/api/shipments', methods=['POST'])
def add_shipment():
    """Add a new shipment."""
    data = request.json
    shipment = ShipmentService.add_shipment(data)
    return jsonify(shipment.to_dict()), 201

@shipment_bp.route('/api/shipments/<int:shipment_id>', methods=['PUT'])
def update_shipment(shipment_id):
    """Update an existing shipment."""
    data = request.json
    shipment = ShipmentService.update_shipment(shipment_id, data)
    if shipment:
        return jsonify(shipment.to_dict()), 200
    return jsonify({'message': 'Shipment not found'}), 404

@shipment_bp.route('/api/shipments/<int:shipment_id>', methods=['DELETE'])
def delete_shipment(shipment_id):
    """Delete a shipment."""
    success = ShipmentService.delete_shipment(shipment_id)
    if success:
        return jsonify({'message': 'Shipment deleted'}), 200
    return jsonify({'message': 'Shipment not found'}), 404

File: MERN_Permacore/server/src/api/routes/shipmentRoutes.js
const express = require('express');
const router = express.Router();

// Import the shipment controller
const shipmentController = require('../controllers/shipmentController');

// Middleware for authentication - Placeholder (uncomment and adjust according to your auth system)
// const { isAuthenticated } = require('../middleware/authMiddleware');

// Middleware for validating request payloads - Placeholder
// const validateShipment = require('../middleware/validateShipmentMiddleware');

router.get('/openShipments', shipmentController.getOpenShipments);
// GET all shipments
// Adding isAuthenticated middleware as an example of how to protect a route
router.get('/', /* isAuthenticated, */ shipmentController.getAllShipments);


// POST a new shipment - Including an example of validation middleware
router.post('/', /* isAuthenticated, validateShipment, */ shipmentController.createShipment);

// GET a single shipment by id
// Note: This is moved above dynamic :id routes to prevent overshadowing
router.get('/:id', /* isAuthenticated, */ shipmentController.getShipmentById);

// PUT to update a shipment - Assuming validation would be similar to creation
router.put('/:id', /* isAuthenticated, validateShipment, */ shipmentController.updateShipment);

// DELETE a shipment
router.delete('/:id', /* isAuthenticated, */ shipmentController.deleteShipment);

// New route to check inspection status for a specific shipment
router.get('/:id/inspectionStatus', /* isAuthenticated, */ async (req, res) => {
    try {
        const shipmentId = req.params.id;
        const isInspected = await shipmentController.checkInspectionStatus(shipmentId);
        
        res.status(200).json({ inspected: isInspected });
    } catch (error) {
        console.error('Error checking inspection status:', error);
        res.status(500).json({ message: 'Internal Server Error', error: error.message });
    }
});

module.exports = router;
